### PR TITLE
Camera extrinsics

### PR DESCRIPTION
**Pull request by zpillio**
Adds camera extrinsics to the aligment calculation and the cost function.
Database format is changed! A new column called extrinsics is added. It's a 4x4 transformation matrix from the frame of position prior (GPS) to the camera.

Extrinsics also added as a new output file to be able to read in the alignment step. Output is separated to keep the compatibility with other programs which could read colmap output.

### CODE REVIEW

File: src/exe/colmap.cc [hunk 1]
```diff
@@ -1182,7 +1182,9 @@ int RunModelAligner(int argc, char** argv) {
     for (size_t i = 0; i < ref_image_names.size(); ++i) {
       const Image* image = reconstruction.FindImageWithName(ref_image_names[i]);
       if (image != nullptr) {
-        errors.push_back((image->ProjectionCenter() - ref_locations[i]).norm());
+        const class Camera& camera = reconstruction.Cameras().at(image->CameraId());
```
Comment (DeinFreund) [inline src/exe/colmap.cc:1185]: It's a bit weird now that model_aligner reads the priors/alignment positions from a text file but reads the extrinsics from the database.

File: src/base/cost_functions.h [hunk 2]
```diff
@@ -57,15 +57,16 @@ class GpsPriorCostFunction {
...
+    residuals[0] = costFactorLatLon * (camPosRot[0] + tvec[0] - posPriorPos[0]);
+    residuals[1] = costFactorLatLon * (camPosRot[1] + tvec[1] - posPriorPos[1]);
+    residuals[2] = costFactorAlt * (camPosRot[2] + tvec[2] - posPriorPos[2]);
 
     return true;
   }
 
  private:
     const Eigen::Vector3d posPrior;
+    const Eigen::Vector3d posPriorPos;
```
Comment (DeinFreund) [inline src/base/cost_functions.h:69]: might be a bit clearer to call it `posPriorOffset` or `posPriorExtrinsics`

File: src/exe/colmap.cc (hunk 1 above)
Comment (zpillio) [reply src/exe/colmap.cc:1185]: It doesn't read the database.
Mapper writes out with WriteCamerasExtrinsics to a separated file, and model_aligner read them.

**Comment by zpillio** [comment]
> Impressive. Does it converge well and fix the offset issue?

If I'm using large cost_factor which makes the alignment error small. The position error in VGPS is less than 1m. So yes, I think now the two parts of the system are consistent.

File: src/base/cost_functions.h (hunk 2 above)
Comment (zpillio) [reply src/base/cost_functions.h:69]: I will do

File: src/exe/colmap.cc (hunk 1 above)
Comment (DeinFreund) [reply src/exe/colmap.cc:1185]: You're right.

**Pull request merged**